book_main.rb

```diff
@@ -0,0 +1,17 @@
+require_relative './book'
+
+class Bookmain
+  def list_all_books(books)
+    books.each { |n| puts n.to_string }
+  end
+
+  def create_a_book(books)
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
```

Comment:
- [ ] All requests for user inputs and validations should be handled by another method in keeping with the single-responsibility rule. Kindly extract these requests here into another method.

  **SUGGESTION:** You can extract these requests methods into another method and have that method return and array of title and author. Then destructure the variables in create_a_book like so:

  ![imuba-books](https://user-images.githubusercontent.com/36542986/157121658-a41c03c9-8b5d-49c2-b448-8bf3bc86e6b4.jpg)